Content/posts/how-we-do-vapor-releases.md

```diff
@@ -0,0 +1,51 @@
+---
+date: 2025-08-14 18:00
+description: See how we approach releases in Vapor, so we can release quickly and often to save time and provide a better experience for users and contributors.
+tags: Framework, Automation
+author: Tim
+authorImageURL: /author-images/tim.jpg
+---
+# How we approach releases at Vapor
+
+I've had some interesting conversations recently at why we do releases the way we do in Vapor so thought it would be worth sharing. 
+
+## Vapor Releases
+
+Our release process is generally pretty simple. For everything outside of new major versions we treat each pull request as a new release. Each pull request will get tagged with one of three labels:
+
+* `no-release-needed` - this usually encompasses things like test changes, updates to CI or the README or updates to governance files. Basically anything that doesn't touch `Sources` and wouldn't result in any changes for people pulling in the code. 
+* `semver-patch` - in following with SemVer guidelines this would be bug fixes that don't have an effect on the public API. 
+* `semver-minor` - changes that would introduce new APIs that can be adopted. 
+
+The pull request will go through a review by one of the maintainers/core team, and obviously must pass all the tests in CI across the different platforms and Swift versions we support. Once that's all green, we merge the PR and use [Penny](), Vapor's bot, to create a release. The pull request title and body form the release title and body (which is why you might see us edit them if you submit a PR) and if the PR is tagged with SemVer patch or minor, Penny works out the release number, tags the code, creates a release in GitHub and notifies Discord. The only exception to this is security releases because GH doesn't support automations on security forks yet.
```

Comment:
```suggestion
The pull request will go through a review by one of the maintainers/core team, and obviously must pass all the tests in CI across the different platforms and Swift versions we support. Once that's all green, we merge the PR and use [Penny](https://github.com/vapor/penny-bot), Vapor's bot, to create a release. The pull request title and body form the release title and body (which is why you might see us edit them if you submit a PR) and if the PR is tagged with SemVer patch or minor, Penny works out the release number, tags the code, creates a release in GitHub and notifies Discord. The only exception to this is security releases because GH doesn't support automations on security forks yet.
```